tensorflow/compiler/xla/shape_util.cc

```diff
@@ -654,6 +654,12 @@ ShapeUtil::MakeShapeWithDescendingLayoutAndSamePhysicalLayout(
       .IgnoreLayout()(lhs, rhs);
 }
 
+/* static */ bool ShapeUtil::CompatibleIgnoringElementTypeAndDimensions(
```

Comment:
How about CompatibleKind? Since this can can only return false if
1. kind_ disagrees
2. Tuple has different element count or nesting

Reply:
Seems reasonable, the name was unnecessarily long and the comment describes what it does more specifically anyway. I'll update the pr.